### PR TITLE
Get total roles count from DB

### PR DESCRIPTION
**Pull request by sahandilshan**
### Proposed changes in this pull request

* Add support for RoleDAO to get the role count from the DB.
* Fix https://github.com/wso2/product-is/issues/13823

### CODE REVIEW

File: components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java [hunk 1]
```diff
@@ -1482,6 +1512,28 @@ public Set<String> getSystemRoles() {
         return systemRoles;
     }
 
+    @Override
+    public int getRolesCount(String tenantDomain) throws IdentityRoleManagementException {
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
```
Comment (ashensw) [inline components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java:1517]: Add a new line after the method signature.

File: components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAO.java [hunk 2]
```diff
@@ -241,4 +241,13 @@ default Set<String> getSystemRoles() throws NotImplementedException {
...
         throw new NotImplementedException("getSystemRoles method is not implemented");
     }
+
+    /**
+     * Retrieve the count of tenant roles.
+     *
+     * @param tenantDomain tenant domain.
+     */
+    default int getRolesCount(String tenantDomain) throws IdentityRoleManagementException {
+        throw new NotImplementedException("getRoleCount method is not implemented");
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAO.java:252]: Fixed with cd84a4114fc20acf8dbd6155122c658d45aebd46

File: components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementService.java [hunk 3]
```diff
@@ -203,4 +203,13 @@ default Set<String> getSystemRoles() throws NotImplementedException {
...
         throw new NotImplementedException("getSystemRoles method is not implemented");
     }
+
+    /**
+     * Retrieve the count of tenant roles.
+     *
+     * @param tenantDomain tenant domain.
+     */
+    default int getRolesCount(String tenantDomain) throws IdentityRoleManagementException {
+        throw new NotImplementedException("getNumberOfRoles method is not implemented");
```
[the remark this answers is not in the document]
Comment (sahandilshan) [reply components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementService.java:214]: Fixed with cd84a4114fc20acf8dbd6155122c658d45aebd46

File: components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java (hunk 1 above)
Comment (sahandilshan) [reply components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java]: Fixed with cd84a4114fc20acf8dbd6155122c658d45aebd46

File: components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java [hunk 4]
```diff
@@ -1482,6 +1512,28 @@ public Set<String> getSystemRoles() {
...
     }
 
+    @Override
+    public int getRolesCount(String tenantDomain) throws IdentityRoleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        try (Connection connection = IdentityDatabaseUtil.getUserDBConnection(false)) {
+            String databaseProductName = connection.getMetaData().getDatabaseProductName();
+            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                    getDBTypeSpecificRolesCountQuery(databaseProductName), RoleTableColumns.UM_ID)) {
```
Comment (ashensw) [inline components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java:1522]: Do we need to pass a returning column name for NamedPreparedStatement here?

Comment (sahandilshan) [reply components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java]: Fixed with 62f9bb64b2fdb14ee53e0a37cb1d9e7bf33a70d7

**Review by ashensw — APPROVED** [review]

**Pull request merged**